### PR TITLE
Changes Magic Mirror Race Blacklist to use species blacklist

### PR DESCRIPTION
**Pull request by Robustin**
:cl: Robustin
balance: Magic mirrors have been updated to use the existing species blacklist system. 
/:cl:

### CODE REVIEW

File: code/game/objects/structures/mirror.dm [hunk 1]
```diff
@@ -89,15 +89,14 @@
...
-	var/list/races_blacklist = list("skeleton", "agent", "angel", "military_synth", "memezombies")
 	var/list/choosable_races = list()
 
 /obj/structure/mirror/magic/New()
 	if(!choosable_races.len)
 		for(var/speciestype in subtypesof(/datum/species))
 			var/datum/species/S = new speciestype()
-			if(!(S.id in races_blacklist))
-				choosable_races += S.id
+			if(!S.dangerous_existence || !S.blacklisted)
```
Comment (ChangelingRain) [inline code/game/objects/structures/mirror.dm]: rip plasmaman wizard

File: code/game/objects/structures/mirror.dm [hunk 2]
```diff
@@ -89,14 +89,13 @@
...
 	icon_state = "magic_mirror"
-	var/list/races_blacklist = list("skeleton", "agent", "angel", "military_synth", "memezombies")
 	var/list/choosable_races = list()
 
 /obj/structure/mirror/magic/New()
 	if(!choosable_races.len)
 		for(var/speciestype in subtypesof(/datum/species))
 			var/datum/species/S = new speciestype()
-			if(!(S.id in races_blacklist))
+			if(!S.dangerous_existence && !S.blacklisted)
```
Comment (ChangelingRain) [inline code/game/objects/structures/mirror.dm]: rip plasmaman wizard

**Review by vuonojenmustaturska — DISMISSED** [review]
I'd say this is more of a fix really

**Comment by Buggy123** [comment]
Does this actually change which species currently are/aren't available?

**Comment by Robustin** [comment]
Yes, significantly. Limits you to the mostly fluffier races like podpeople, slimepeople, moths, lizards, etc.

**Comment by Buggy123** [comment]
I don't agree with this then, as-is I think magic mirrors are fairly balanced. The Pride's mirror on lavaland spaces you, forcing you to bring protection or healing and a way to get back. The wizard gets to use one for free, but wizards are _wizards_.

All the _really_ dangerous or astoundingly superior races like infectious zombies and combat synths have been removed from the mirror already. For wizards none of the options are a particularly huge power boost. Golems are nice but generally slow and can't wear armor, the rest is pretty gimmicky or situational.

Finally, _you didn't mention this stuff in the PR or the title!_ I had to _ask_ if this is just a minor code change or a substantial balance change, and judging from the sheer lack of discussion here it seems like no one else got the memo either. At minimum please update the title and PR, maybe even close/reopen it so people ingame get a alert about this as well.

**Comment by OmegaShoots** [comment]
This isn't a refactor, it's a balance PR

**Comment by AlexTheSysop42** [comment]
what I thought this was an optimization but it's a balance change reeeeeee

@ChangelingRain TAG THIS WITH REBALANCE

**Comment by Buggy123** [comment]
If this goes through without at least a semblence of proper discussion I will immediately make a PR reverting it with a big, obvious, "REVERTS POORLY DOCUMENTED MAGIC MIRROR BALANCE CHANGES DISGUISED AS A CODE REFACTOR" title.

for shame robustin, for shame.

**Comment by Jordie0608** [comment]
>refactors

**Comment by AlexTheSysop42** [comment]
there's already a weaker mirror only for lighter races, why do we need to change the normal to that as well?

**Comment by PKPenguin321** [comment]
im just tired of wizards turning themselves into zombies and needing me to give them their arms back

**Comment by AlexTheSysop42** [comment]
Just don't give them their arms, it's their fault they picked that race

**Comment by PKPenguin321** [comment]
Yeah I'll just let the round become pseduo extended every time a wizard spawns good idea

**Comment by coiax** [comment]
The vars on species datums need to be cleaned up. Too many things relying on poorly named variables

**Comment by Kromgar** [comment]
@PKPenguin321 I thought claw zombies got removed from the magic mirror

**Comment by Buggy123** [comment]
They did.

**Comment by AlexTheSysop42** [comment]
oh I thought pk was talking about wizard choosing high-functioning zombies and then losing both arms

**Comment by Robustin** [comment]
@Buggy123 

For every PR that removes Combat Synths we get a new one adding shit like clockwork golems. It's never going to end until we just uses the species tags as intended.

There's absolutely nothing "balanced" about what was once a purely cosmetic fluff piece turning into the most important wizard item at absolutely no cost. We pulled SKELETONS from the mirror initially because one of the big draws of being a lich was being spaceproof (gem suit) and chem immune without sacrifice speed. Now skeletons wouldn't even make a top 5 list of the bullshit we have in there.

Clockwork golems: Heavier/40% stun punches, free innate armor, SPECIES_INORGANIC, NOBREATH, NOZOMBIE, RADIMMUNE, NOBLOOD, RESISTCOLD, RESISTPRESSURE, PIERCEIMMUNE, oh and shockproof. 

Not to mention all the shit you can get away with using non-infectious zombie (free regen), vampire, synth, etc. I could just keep adding to the snowflake list but every time we add some ridiculous new species nobody is going to remember "oh I should add this to that one list in the magic mirror file".

**Comment by ChangelingRain** [comment]
yeah but you're also removing plasmamen, who are a bad wizard meme at best.

**Comment by AlexTheSysop42** [comment]
@Robustin Why has no one (including you) commented on every op race PR saying "add this to the mirror file"?

**Comment by Robustin** [comment]
Thats a joke right Alex?

**Comment by Durkel** [comment]
Every single wizard round the wizard turns them self into a dullahan then ahelps when they can't turn back. Save me for this nothing I've become, Robustin.

**Comment by AlexTheSysop42** [comment]
PRs do have discussion for a reason, if this is that big of a problem then someone should bring it up.

Or we could just put a comment in the base race file warning about this.

**Comment by Robustin** [comment]
Sticking to my guns on this one. This issue is going to repeat itself for all eternity until something is done. Nobody is going to actually keep this snowflake list properly maintained and why would they when we already have a list of dangerous/blacklisted races.

**Comment by ExcessiveUseOfCobblestone** [comment]
This is good for OOing the blacklist on the species, I'm just wondering if it would be beneficial to separate general blacklist from mirror blacklist.

**Comment by ExcessiveUseOfCobblestone** [comment]
actually nvm if you're going to separate them then it would be more sensible to have the mirror-only blacklist on the mirror.

**Comment by duncathan** [comment]
@KorPhaeron @Cyberboss

**Comment by Robustin** [comment]
This already got a bandaid fix to the most abused race so I figure we're just gonna keep the cycle going since most wizards die within 5 minutes regardless of this awful mechanic.

**Pull request closed without merging**